From ab2fd3fe79c19462cd54ffe36792b8d729e32541 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Thu, 25 Oct 2012 15:55:29 +0300
Subject: [PATCH 08/10] Use gold linker if available

Note that the GCC 4.7 syntax is left commented out for future
convenience.

Later additions:
Force SySV hash style instead of GNU style. This is a desperate attempt
to somehow stay below the 3.2GB memory limit for linking.
---
 Tools/qmake/mkspecs/features/unix/default_post.prf |   13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/Tools/qmake/mkspecs/features/unix/default_post.prf b/Tools/qmake/mkspecs/features/unix/default_post.prf
index 82e254e..9ec3b05 100644
--- a/Tools/qmake/mkspecs/features/unix/default_post.prf
+++ b/Tools/qmake/mkspecs/features/unix/default_post.prf
@@ -33,15 +33,16 @@ linux-g++*:isEqual(QT_ARCH,i386) {
 contains(TEMPLATE, app): CONFIG += rpath
 
 isEqual(QT_ARCH,i386):CONFIG(debug, debug|release) {
-  # Make ld don't cache the symbol tables of input files in memory to avoid memory exhaustion during the linking phase.
-  # We have to use ld, because --no-keep-memory isn't supported by ld.gold.
+  # Make ld not cache the symbol tables of input files in memory to
+  # avoid memory exhaustion during the linking phase.
   QMAKE_LFLAGS += -Wl,--no-keep-memory
-} else {
   # Use gold if available
-  !scratchbox:!mac:exists(/usr/bin/ld.gold) {
-      # Upstream gcc 4.7 does not support the -fuse-ld=gold option ( see: http://sourceware.org/ml/binutils/2011-01/msg00178.html )
-      isEqual(QT_GCC_MAJOR_VERSION, 4):lessThan(QT_GCC_MINOR_VERSION, 7):QMAKE_LFLAGS += -fuse-ld=gold
+  exists(/usr/bin/ld.gold) {
+      # Gah. use-ld=gold is GCC 4.7+
+      #QMAKE_LFLAGS += -fuse-ld=gold
   }
+  # For the love of all that is good, even gold runs out of memory
+  QMAKE_LFLAGS += -Wl,--hash-style=sysv
 }
 
 load(default_post)
-- 
1.7.10.4

